okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -1050,6 +1035,99 @@ private ModuleDescriptor findSystemInterface(Tenant tenant, String interfaceName
     return promise.future();
   }
 
+  private boolean isExclusive(ModuleDescriptor md) {
+    return md.getSystemInterface("_tenantPermissions") != null
+        || md.getAuthRoutingEntry() != null;
+  }
+
+  private boolean depsOK(TenantModuleDescriptor tm, ModuleDescriptor md,
+      Collection<ModuleDescriptor> modules, Set<String> allProvided) {
+
+    if (tm.getAction().equals(Action.disable)) {
+      return DepResolution.moduleDepRequired(modules, md);
+    }
+    return DepResolution.moduleDepProvided(modules, allProvided, md);
+  }
+
+  private void jobRunPending(Tenant t, ProxyContext pc, TenantInstallOptions options,
+      List<TenantModuleDescriptor> tml, Map<String, ModuleDescriptor> modsAvailable,
+      Set<String> allProvided, InstallJob job, AtomicInteger running, AtomicBoolean exclusive,
+      Promise<Void> promise) {
+
+    Iterator<TenantModuleDescriptor> iterator = tml.iterator();
+    while (iterator.hasNext() && running.get() < options.getMaxParallel() && !exclusive.get()) {
+      TenantModuleDescriptor tm = iterator.next();
+      ModuleDescriptor md = modsAvailable.get(tm.getId());
+      if ((tm.getStage().equals(TenantModuleDescriptor.Stage.pending)
+          || tm.getStage().equals(TenantModuleDescriptor.Stage.deploy))
+          && depsOK(tm, md, getEnabledModules(t), allProvided)
+          && (!isExclusive(md) || running.get() == 0)) {
```

Comment:
This complex if with 5 expressions combined with `|| && && && ||` can be simplified by using guard clauses; this also reduces indentation of the body of the if:

```
      if (   tm.getStage() != TenantModuleDescriptor.Stage.pending
          && tm.getStage() != TenantModuleDescriptor.Stage.deploy ) {
        continue;
      }
      if (! depsOK(tm, md, getEnabledModules(t), allProvided)) {
        continue;
      }
      ...
```